grammar/basic-grammar.xml

```diff
@@ -320,6 +320,25 @@
     <non-terminal ref="SymbolicName" rr:title="function name"/>
   </production>
 
+  <production name="ExplicitProcedureInvocation">
+    <non-terminal ref="ProcedureName"/>
+    &WS; ( &WS;
+    <opt>&expr;<repeat>, &WS; &expr;</repeat></opt>
+    &WS; )
+  </production>
+
+  <production name="ImplicitProcedureInvocation">
+    <non-terminal ref="ProcedureName"/>
+  </production>
+
+  <production name="ProcedureResultField" rr:inline="true">
+    <non-terminal ref="SymbolicName" rr:title="procedure result field"/>
+  </production>
+
+  <production name="ProcedureName" rr:inline="true">
+    <non-terminal ref="SymbolicName" rr:title="procedure name"/>
+  </production>
+
```

Comment:
We sped up procedures work based on our preliminary extensibility discussion which made me feel like it belonged into the core. Let's revisit this later.